cadence/templates/server-configmap.yaml

```diff
@@ -152,26 +152,21 @@ data:
         {{- end}}
 
     clusterMetadata:
-      enableGlobalDomain: {{ `{{ default .Env.ENABLE_GLOBAL_DOMAIN "true" }}` }}
-      failoverVersionIncrement: 10
-      masterClusterName: "primary"
-    {{- if `{{ .Env.IS_NOT_PRIMARY }}` }}
-      currentClusterName: "secondary"
-    {{- else }}
-      currentClusterName: "primary"
-    {{- end }}
+      enableGlobalDomain: {{ .Values.server.config.clusterMetadata.enableGlobalDomains }}
+      failoverVersionIncrement: {{ .Values.server.config.clusterMetadata.maximumClusterCount }}
+      masterClusterName: {{ .Values.server.config.clusterMetadata.masterClusterName }}
+      currentClusterName: {{ .Values.server.config.clusterMetadata.currentClusterName }}
       clusterInformation:
-        primary:
-          enabled: true
-          initialFailoverVersion: 0
-          rpcName: "cadence-frontend"
-          rpcAddress: {{ `{{ default .Env.PRIMARY_FRONTEND_SERVICE "cadence" }}` }}:{{ .Values.server.frontend.service.port | default `{{ default .Env.FRONTEND_PORT 7933 }}` }}
-      {{- if `{{ .Env.ENABLE_GLOBAL_DOMAIN }}` }}
-        secondary:
-            enabled: true
-            initialFailoverVersion: 2
-            rpcName: "cadence-frontend"
-            rpcAddress: {{ `{{ default .Env.SECONDARY_FRONTEND_SERVICE "cadence-secondary" }}` }}:{{ .Values.server.frontend.service.port | default `{{ default .Env.FRONTEND_PORT 7933 }}` }}
+      {{- $frontendComponentName := (include "cadence.componentname" (list . "frontend")) }}
+      {{- $serverFrontendServicePort := .Values.server.frontend.service.port }}
+      {{- $clusterIndex := 0 }}
+      {{- range $clusterName, $clusterInfo := .Values.server.config.clusterMetadata.clusterInformation }}
```

Comment:
Is it ranging over a Golang map? Would it be a problem because iterating map in Golang is not deterministic. 

May we should change it to use List in `.Values.server.config.clusterMetadata.clusterInformation` instead

Reply:
The map is iterated in the key-sorted order it is defined in text/templates.
[Source](https://pkg.go.dev/text/template#hdr-Actions).
> {{range pipeline}} T1 {{end}}
	The value of the pipeline must be an array, slice, map, or channel.
	If the value of the pipeline has length zero, nothing is output;
	otherwise, dot is set to the successive elements of the array,
	slice, or map and T1 is executed. If the value is a map and the
	keys are of basic type with a defined order, the elements will be
	visited in sorted key order.

I did account for that (after starting to change it to an array, but didn't like it due to possible unwanted name collisions).
Then I read the docs and decided to go with the map, because of the sorting.

But now that I'm explaining it I actually realized if the user adds a Cadence cluster to an existing group which does not sort at the end (for example having a cluster named `master` and adding one named `alternative` later) messes up the failover versions and multiple clusters may end up with the same failover version, so let me change it to array instead.

Reply:
Fixed in https://github.com/banzaicloud/banzai-charts/compare/e62679633c3cd17e00bb255b1487d508a01f13cf..d7a21e7928d6a552bb021bfc742df24012f6afe7

Reply:
Tested the array with XDC using a `LoadBalancer`, it worked.
I haven't checked the failover (it was unfeasible in my test environment), but based on the config it should work as expected.

```yaml
server:
# ...
  config:
    clusterMetadata:
      enableGlobalDomain: true
      maximumClusterCount: 100
      masterClusterName: "cluster-0"
      currentClusterName: "cluster-1" # Note: was cluster-0 on the first K8s cluster, cluster-1 on the second.
      clusterInformation:
        - name: cluster-0
          enabled: true
          rpcAddress: aaad5b62308f440019a0c5d4db7ca8e5-878378017.eu-central-1.elb.amazonaws.com:7933
        - name: cluster-1
          enabled: true
          rpcAddress: a96b3ac0a354142d892bf9cec6d8632c-822865977.eu-central-1.elb.amazonaws.com:7933
# ...
  frontend:
    service:
      type: LoadBalancer
# ...
```

```shell
kubectl --kubeconfig "${TEST_KUBECONFIG}" --namespace "${TEST_NAMESPACE}" logs cadence-worker-78c4cc7b54-mk8ff
```

```json
...
{"level":"info","ts":"2021-08-17T09:01:24.377Z","msg":"Add new peers by DNS lookup","service":"cadence-worker","address":"a96b3ac0a354142d892bf9cec6d8632c-822865977.eu-central-1.elb.amazonaws.com","addresses":"[3.69.86.201:7933 18.197.78.25:7933 3.123.162.100:7933]","logging-call-at":"clientBean.go:321"}
{"level":"info","ts":"2021-08-17T09:01:24.377Z","msg":"Add new peers by DNS lookup","service":"cadence-worker","address":"aaad5b62308f440019a0c5d4db7ca8e5-878378017.eu-central-1.elb.amazonaws.com","addresses":"[3.69.79.22:7933 35.156.195.186:7933 18.156.94.225:7933]","logging-call-at":"clientBean.go:321"}
...
```